engine/access/rest/converter.go

```diff
@@ -73,39 +169,80 @@ func toTransactionSignature(transactionSignature *generated.TransactionSignature
 		return flow.TransactionSignature{}, err
 	}
 
+	signerIndex, err := toUint64(transactionSignature.SignerIndex)
+	if err != nil {
+		return flow.TransactionSignature{}, err
+	}
+
+	keyIndex, err := toUint64(transactionSignature.KeyIndex)
+	if err != nil {
+		return flow.TransactionSignature{}, err
+	}
+
 	return flow.TransactionSignature{
 		Address:     address,
-		SignerIndex: int(transactionSignature.SignerIndex),
-		KeyIndex:    uint64(transactionSignature.KeyIndex),
+		SignerIndex: int(signerIndex),
+		KeyIndex:    keyIndex,
 		Signature:   signature,
 	}, nil
 }
 
 func toTransactionSignatures(sigs []generated.TransactionSignature) ([]flow.TransactionSignature, error) {
+	if len(sigs) == 0 {
+		return nil, nil
+	}
 	signatures := make([]flow.TransactionSignature, len(sigs))
-	for _, sig := range sigs {
+	for x, sig := range sigs {
```

Comment:
because of my humble c++ beginnings are creeping out, where people were happy about 'x's, you could even say they were encouraging them :D will fix tho